[do not review] aggressive consistency checks & fatal on stats mismatch

tbg opened the pull request:
I just need a ref for TC.

Release note: None

tbg committed:
aggressive consistency checks & fatal on stats mismatch

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/41444)
<!-- Reviewable:end -->

tbg committed:
storage: remove gunk in TestStore

This was added in #17021 to regression test a bug regarding AddSSTable
and preemptive snapshots. At this point, this random insertion in an
unrelated test is unfortunate, so remove it.

I caught this when I ran with aggressive consistency checks; the SST
ingestion messes up the stats.

Release note: None

tbg committed:
cli: preserve MVCCStats in unsafe-remove-dead-replicas

It's easy enough to do, so let's do it.

Release note: None

tbg committed:
aggressive

tbg committed:
skips

tbg committed:
roachtest: set acceptance/version-upgrade timeout to 30m

When adding aggressive consistency checks, I noticed that
acceptance/version-upgrade got pushed over the limit.

Release note: None
